lib/src/model/activity.dart

```diff
@@ -0,0 +1,107 @@
+enum ActionType {
+  actionUnknonwn,
+  actionCreateRecipe,
+  actionMint,
+  actionPurchase,
+  // further types
+}
+
+extension AccountActionTypeDePar on ActionType {
+  String itemToString() {
+    final _stringMap = <String, ActionType>{
+      'unknown': ActionType.actionUnknonwn,
+      'created': ActionType.actionCreateRecipe,
+      'minted': ActionType.actionMint,
+      'purchased': ActionType.actionPurchase,
+    };
+
+    return _stringMap.keys.firstWhere((key) => _stringMap[key] == this, orElse: () => 'unknown');
+  }
+}
+
+
+extension AccountActionTypePar on String {
+  ActionType fromString() {
+    final _stringMap = <String, ActionType>{
+      'unknown': ActionType.actionUnknonwn,
+      'created': ActionType.actionCreateRecipe,
+      'minted': ActionType.actionMint,
+      'purchased': ActionType.actionPurchase,
+    };
+
+    return _stringMap[this] ?? ActionType.actionUnknonwn;
+  }
+}
+
+
+class Activity {
```

Comment:
This file is db related and is not even used in wallet. Why sdk need it?

Reply:
i wasn't checking everything thoroughly! i was moving everything en masse w/ the assumption we'd worry about making it not suck later, b/c anything that sucks here already sucked. we're just moving the suck around. sure, put it on the kill list.